src/Microsoft.Azure.SignalR.Management/ServiceManager.cs

```diff
@@ -160,27 +158,15 @@ public string GenerateClientAccessToken(string hubName, string userId = null, IL
 
         public async Task<bool> IsServiceHealthy(CancellationToken cancellationToken)
         {
-            var isHealthy = false;
-            var api = await _restApiProvider.GetServiceHealthEndpointAsync();
-            await _restClient.SendAsync(api, HttpMethod.Get, _productInfo, handleExpectedResponse: response =>
-                {
-                    switch (response.StatusCode)
-                    {
-                        case HttpStatusCode.OK:
-                            isHealthy = true;
-                            return true;
-                        case HttpStatusCode.BadGateway:
-                            isHealthy = false;
-                            return true;
-                        case HttpStatusCode.ServiceUnavailable:
-                            isHealthy = false;
-                            return true;
-                        default:
-                            return false;
-                    }
-                },
-                cancellationToken: cancellationToken);
-            return isHealthy;
+            var healthApi = _restClient.HealthApi;
+            var httpOperationResponse = await healthApi.GetHealthStatusWithHttpMessagesAsync(cancellationToken: cancellationToken);
```

Comment:
the original code has some azure exceptions like this
```
            throw response.StatusCode switch
            {
                HttpStatusCode.BadRequest => new AzureSignalRInvalidArgumentException(response.RequestMessage.RequestUri.ToString(), innerException, detail),
                HttpStatusCode.Unauthorized => new AzureSignalRUnauthorizedException(response.RequestMessage.RequestUri.ToString(), innerException),
                HttpStatusCode.NotFound => new AzureSignalRInaccessibleEndpointException(response.RequestMessage.RequestUri.ToString(), innerException),
                _ => new AzureSignalRRuntimeException(response.RequestMessage.RequestUri.ToString(), innerException),
            };
```
Is it possible to config the exceptions for autorest client? otherwise it's a breaking change.

Reply:
According to Liangying, this breaking change is acceptable.

Reply:
does our Function output binding depend on these exceptions @zackliu? If not, is changing this exception to some general exceptions/not throw acceptable? cc @vwxyzh

Reply:
@vicancy It's designed for helping customer self-troubleshooting, they can catch AzureSignalRException, and the exceptions listed are having messages include some mitigate suggestions that customers can task action. Function output binding doesn't depend on them.